### ISSUE TITLE
[.Net] ExecuteScriptAsync / PostWebMessageAsJson / PostWebMessageAsString with buffer - not string

### ISSUE THREAD

**Issue by kirsan31** [MicrosoftEdge/WebView2Feedback#2193]
**Is your feature request related to a problem? Please describe.**

With a certain frequency (~ 2 seconds), we need to execute some rather large scripts in WebView2. Because `ExecuteScriptAsync` only takes a `string` as a parameter, and the `string` in .Net is an immutable type, then we can't use a buffer for this. As a result every time we need to create a new one, and most of the time the garbage collector collects these strings, which naturally leads to lags (not visual of course, but significant for us) :(

**Describe the solution you'd like and alternatives you've considered.**

Create an overload of `ExecuteScriptAsync` / `PostWebMessageAsJson` / `PostWebMessageAsString` that will take some buffer as input parameter (`ReadOnlyMemory<char>` / `char[]`). For example:
```cs
public System.Threading.Tasks.Task<string> ExecuteScriptAsync (ReadOnlyMemory<char> script);
```
May be some other workaround exists that I missed?

[AB#38225871](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/38225871)

**Comment by champnic** [comment]
Thanks for the feature request @kirsan31! I'll add this to our backlog.

Have you considered using [AddScriptToExecuteOnDocumentCreatedAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addscripttoexecuteondocumentcreatedasync?view=webview2-dotnet-1.0.1108.44) to add scripts to the page, and then the ExecuteScriptAsync would just call one of the scripts? Or you could consider loading a script from file (although not sure if that works from a performance perspective for your scenario.

**Comment by kirsan31** [comment]
@champnic 
> Have you considered using [AddScriptToExecuteOnDocumentCreatedAsync](https://docs.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addscripttoexecuteondocumentcreatedasync?view=webview2-dotnet-1.0.1108.44) to add scripts to the page, and then the ExecuteScriptAsync would just call one of the scripts?

WebView2 displays 3D Scatter Plotly graph. The script is simple `Plotly.restyle` call and the most significant (in term of string size) part of it is data points (which are calculated on the fly). So you suggestion unfortunately doesn't help :(

> Or you could consider loading a script from file (although not sure if that works from a performance perspective for your scenario.

How will this help? I still need to get the `string` from this file to call the script :(

**Comment by champnic** [comment]
Aha - you are trying to pass a large amount of data, not a bunch of scripting. That makes sense. I think #1005 is tracking the same scenario, so we can use that to track the request. I'll try to bump the priority.

I'm not familiar with the specifics, but one dev noted using binary data and ArrayBuffers. Could something like that work for you?

**Comment by kirsan31** [comment]
@champnic no, https://github.com/MicrosoftEdge/WebView2Feedback/issues/1005 is a different case.

#1005:
- Common (all frameworks) WebView2 problem.
- It's about big data (terabytes) transfer and performance of it.
- It's about web apps (where you can decide how to receive and process data).
- To solve this problem you need to introduce new API.

Our case:
- it's only .Net problem due to `string` type. c++ doesn't have it.
- We have no _big_ data in terms of transferring. All scripts are 10 - 200 kB. We send them every ~2 sec and we are ok with transfer performance. The problem is .Net GC constantly need to clean previous versions of these strings.
- We have **very** simple JS script with Plotly in WebView2 and use [Plotly.restyle](https://plotly.com/javascript/plotlyjs-function-reference/#plotlyrestyle) to update it. So with previous point in mind we don't need to overcomplicate it with binary data processing.
- Our problem **can be easily solved** without introducing whole new WebView2 api. All you need is to add an overload to `ExecuteScriptAsync` / `PostWebMessageAsJson` / `PostWebMessageAsString` .Net wrappers with some buffer `ReadOnlyMemory<char>` or `char[]`. Like with [`TextWriter.WriteAsync`](https://docs.microsoft.com/en-us/dotnet/api/system.io.textwriter.writeasync?view=net-6.0).

Overall thought at the end: .Net `ExecuteScriptAsync` / `PostWebMessageAsJson` / `PostWebMessageAsString` -> native WebView2 code without `string` allocation.

**Comment by champnic** [comment]
@kirsan31 Thanks for those details! This does sound a lot less complex, and something that should be easy to add. I've reopened it!
